src/ripple/ledger/impl/PaymentSandbox.cpp

```diff
@@ -260,4 +262,138 @@ PaymentSandbox::apply (PaymentSandbox& to)
     tab_.apply(to.tab_);
 }
 
+std::pair<std::map<std::tuple<AccountID, AccountID, Currency>, STAmount>, XRPAmount>
+PaymentSandbox::balancesChanges (ReadView const& view) const
+{
+    using key = std::tuple<AccountID, AccountID, Currency>;
+    // Map of delta trust lines
+    // As a special case, when both ends of the trust line are the same currency, then
+    // it's delta currency for that issuer
+    // To get the change in XRP balance, Account == root, issuer == root, currency == XRP
+    std::map<key, STAmount> result;
+
+    // populate a dictionary with low/high/currency/delta // this can be
+    // compared with the old flow, and xrp deltas// can be checked with amount
+    // destroyed
+    auto each = [&result](uint256 const& key, bool isDelete,
+        std::shared_ptr<SLE const> const& before,
+        std::shared_ptr<SLE const> const& after) {
+
+        STAmount oldBalance;
+        STAmount newBalance;
+        AccountID lowID;
+        AccountID highID;
+
+        // before is read from prev view
+        if (isDelete)
+        {
+            if (!before)
+                return;
+
+            auto const bt = before->getType ();
+            switch(bt)
+            {
+                case ltACCOUNT_ROOT:
+                    oldBalance = (*before)[sfBalance];
+                    newBalance = oldBalance.zeroed();
+                    lowID = xrpAccount();
+                    highID = (*before)[sfAccount];
+                    break;
+                case ltRIPPLE_STATE:
+                    {
+                        lowID = (*before)[sfLowLimit].getIssuer();
+                        highID = (*before)[sfHighLimit].getIssuer();
+                        oldBalance = (*before)[sfBalance];
+                        highID = (*before)[sfAccount];
```

Comment:
That looks like a copy/paste error. We're deleting here, so `newBalance` is zero, it should be `oldBalance.zeroed()`; As-is  it's a default constructed `STAmount` (which does construct to zero, so it worked, but I just got lucky). Good eyes.